cfspopcon/formulas/scrape_off_layer/reattachment_models.py

```diff
@@ -0,0 +1,97 @@
+        sheath_heat_transmission_factor: :term:`glossary link<sheath_heat_transmission_factor>`
+        neutral_flux_density_factor: [10^23 atoms m^-2 s^-1 Pa^-1] :term:`glossary link<neutral_flux_density_factor>`
+        SOL_power_loss_fraction: :term:`glossary link<SOL_power_loss_fraction>`
+        SOL_momentum_loss_function: :term:`glossary link<SOL_momentum_loss_function>`
+        separatrix_electron_density: [1e19 m^-3] :term:`glossary link<separatrix_electron_density>`
+        target_electron_temp: [eV] :term:`glossary link<target_electron_temp>`
+        q_parallel: [GW/m²] :term:`glossary link<q_parallel>`
+
+    Returns:
+        :term:`target_neutral_pressure` [Pa]
+    """
+    SOL_momentum_loss_fraction = calc_SOL_momentum_loss_fraction(SOL_momentum_loss_function, target_electron_temp)
+    lq_int = lambda_q + 1.64 * target_gaussian_spreading
+    b = lq_int / lambda_q
+
+    term1 = 3.0 / 2.0 * np.sqrt(1 - SOL_power_loss_fraction) / (1 - SOL_momentum_loss_fraction)
+    term2 = np.sqrt(average_ion_mass / 2.0)
+    term3 = ((2 * kappa_e0 * kappa_ez) / (7 * parallel_connection_length)) ** (2.0 / 7.0)
+    term4 = (q_parallel) ** (3.0 / 14.0)
+    term5 = b * sheath_heat_transmission_factor * np.sin(np.radians(target_angle_of_incidence)) / (neutral_flux_density_factor * 10**23)
+
+    p0 = (separatrix_electron_density / term1 / term2 / term3 / term4) ** 2 * term5
+
+    return p0
+
+
+@Algorithm.register_algorithm(return_keys=["reattachment_time"])
+def calc_reattachment_time_henderson(
+    target_neutral_pressure: Unitfull,
+    target_electron_density: Unitfull,
+    major_radius: Unitfull,
+    parallel_connection_length: Unitfull,
+    separatrix_power_transient: Unitfull,
+) -> Unitfull:
+    """Calculates the reattachment time for a detachment front to move to e^-5 * original front location from the target.
+
+    Values are normalized to AUG.  per [Henderson Nucl. Fusion 64 (2024) 066006].
+
+    Args:
+      target_neutral_pressure: [Pa] :term:`glossary link<target_neutral_pressure>`
+      target_electron_density: [1e19 m^-3] :term:`glossary link<target_electron_density>`
+      major_radius: [m] :term:`glossary link<major_radius>`
+      parallel_connection_length: [m] :term:`glossary link<parallel_connection_length>`
+      separatrix_power_transient: [MW] :term:`glossary link<separatrix_power_transient>`
+
+    Returns:
+        :term:`reattachment_time` [s]
+    """
+    term1 = target_neutral_pressure / (2.0 * ureg.Pa)
+    term2 = target_electron_density / (30.0 * ureg.n19)
+    # calculate ionization volume using AUG volume (0.4 m^3) and AUG major radius (1.65m)
+    ionization_volume = major_radius / (1.65 * ureg.m) * (0.4 * ureg.m**3)
+    term3 = ionization_volume / (0.4 * ureg.m**3)
+    term4 = parallel_connection_length / (12.0 * ureg.m)
+    term5 = 2.0 * ureg.MW / separatrix_power_transient
+
+    reattachment_time = 0.09 * ureg.s * term1 * term2 * term3 * term4 * term5
+
+    return reattachment_time
```

Comment:
Do we want to use this equation, or equation 6? From Henderson, "given that the time dependence of P_sep plays a crucial role, the simplified expression is improved by assuming a linear dependence with time, such that P_sep = t P^max_sep / t^max. The superscript ‘max’ indicates the maximum power increase of the transient and the time to reach the maximum."

Reply:
i think we use this equation, as the timescales in the other henderson equations are dependent upon the event (ELM vs pellet fueling fusion spike, vs ICRH spike, etc.)  ultimately the t will be limited by the energy confinment time, so maybe we put some kind of advanced calculation in eventually.  for now let's keep it with the simple formula.